src/hotspot/share/utilities/vmError.cpp

```diff
@@ -717,6 +717,12 @@ void VMError::report(outputStream* st, bool _verbose) {
   address lastpc = nullptr;
 
   BEGIN
+  if (MemTracker::enabled() && NmtVirtualMemory_lock != nullptr && NmtVirtualMemory_lock->owned_by_self()) {
+    // Avoid reentrancy due to mallocs in detailed mode.
+    MemTracker::reduce_tracking_to_summary();
+    // Manually unlock if already holding lock upon entering error reporting.
+    NmtVirtualMemory_lock->unlock();
```

Comment:
Thinking this through some more, I am now unsure about my old advice. I think if we force-unlock the mutex here, there should be no need for dropping the tracking mode to summary. Sorry if I gave conflicting advice before.

So I think you could remove the reduce_tracking call (and its implementation).

Dropping to summary has the disadvantage that it makes the NMT report in the hs-err file look like user ran with summary more active, which may confuse analysts. Force-unlocking is the way to go.

Reply:
Ok I see. I've removed the dropping of the tracking level and just kept the force unlocking.